### ISSUE TITLE
Issues with eye server seeing local web server files (when CLI does??)

### ISSUE THREAD

**Issue by MartyStache** [RubenVerborgh/EyeServer#22]
Hola!

I am sure it is something really simple I am missing. I want to use the api but cannot get it to work as designed with local web server files (port 80). Here are some images that depict what I am experiencing. I'm on an M1 Mac Mini (Monterey) if that makes any difference.

First, I try the sample post provided on the github repo home page in postman (which works!):
<img width="342" alt="image" src="https://user-images.githubusercontent.com/62360524/163245698-d1d07f22-b397-4e22-9025-ff9db894ad11.png">

Then I create a local web server (on port 80), files are accessible by a browser without specifying port:
<img width="324" alt="image" src="https://user-images.githubusercontent.com/62360524/163245883-d4d215f0-6f08-41de-8e1c-cb19ebc4c9eb.png">
<img width="661" alt="image" src="https://user-images.githubusercontent.com/62360524/163245905-9b8c5ab8-41d0-4f20-b805-2df5168aa2b8.png">

Then I try the same post, except this time I point to the files on the local web server:
<img width="461" alt="image" src="https://user-images.githubusercontent.com/62360524/163246056-80e734a9-e868-4704-8ca6-6025dbb707df.png">

Then I tried the same post in a Notebook in Visual Studio with Python requests object:
<img width="352" alt="image" src="https://user-images.githubusercontent.com/62360524/163246202-a3e5eb97-94db-48c9-ab6c-fd81b768ddef.png">

Lastly, (what did work!!!) I gave the eye CLI the same arguments (local web server addresses) to the files and it worked!!

<img width="530" alt="image" src="https://user-images.githubusercontent.com/62360524/163246603-efbb05f9-32a2-473a-bd8e-9e7f8a1c5400.png">

Any guidance on how to get this working as intended would be greatly appreciated!

Thanks!

**Comment by RubenVerborgh** [comment]
Maybe the server does not work with more recent EYE versions, that would be an easy explanation.

**Comment by josd** [comment]
Here on my laptop plus WSL 2 (Windows Subsystem for Linux) EyeServer still runs fine with the latest `EYE v22.0409.1317`
```
$ eyeserver 8000
EYE server running on http://localhost:8000
```
```
$ curl "http://localhost:8000/?data=http://eulersharp.sourceforge.net/2003/03swap/socrates.n3&query=http://eulersharp.sourceforge.net/2003/03swap/socratesF.n3"
@prefix log: <http://www.w3.org/2000/10/swap/log#>.
@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#>.
@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>.
@prefix : <http://www.agfa.com/w3c/euler/socrates#>.
@prefix q: <http://www.w3.org/2004/ql#>.

:Socrates a :Man.
:Socrates a :Mortal.
```
```
$ curl "http://localhost:8000/?data=https://n3.restdesc.org/n3/friends.n3&data=https://n3.restdesc.org/n3/knows-rule.n3&query=https://n3.restdesc.org
/n3/query-all.n3"
@prefix ppl: <http://example.org/people#>.
@prefix foaf: <http://xmlns.com/foaf/0.1/>.

ppl:Cindy foaf:knows ppl:John.
ppl:Cindy foaf:knows ppl:Eliza.
ppl:Cindy foaf:knows ppl:Kate.
ppl:Eliza foaf:knows ppl:John.
ppl:Peter foaf:knows ppl:John.
ppl:John foaf:knows ppl:Cindy.
ppl:Eliza foaf:knows ppl:Cindy.
ppl:Kate foaf:knows ppl:Cindy.
ppl:John foaf:knows ppl:Eliza.
ppl:John foaf:knows ppl:Peter.
```

**Comment by MartyStache** [comment]
Thanks Jos!

I am having success as you depicted below with uris pointing to external
resources on the web.

Uris to files on a web server hosted on the same machine as the eye server
seem to be the issue.

**Comment by bdevloed** [comment]
@MartyStache are you using Docker or some kind of virtualization to run EyeServer?
This would explain why localhost for node (i.e. EyeServer) doesn't map to the same as localhost as your cli.

**Comment by MartyStache** [comment]
Nothing that fancy...

on Mac M1 Monterey...running eye server local on port 8000 and web server
for files local on port 80:

[image: image.png]

**Comment by josd** [comment]
Tried the same here and there is indeed an issue:
Via command line it works all fine
```
$ eye --nope http://localhost/eye/reasoning/socrates/socrates.n3 --query http://localhost/eye/reasoning/socrates/socrates-query.n3
eye --nope http://localhost/eye/reasoning/socrates/socrates.n3 --query http://localhost/eye/reasoning/socrates/socrates-query.n3
EYE v22.0409.1317 josd
SWI-Prolog version 8.5.10-15-g673a8f640
starting 32 [msec cputime] 29 [msec walltime]
#Processed by EYE v22.0409.1317 josd
#eye --nope http://localhost/eye/reasoning/socrates/socrates.n3 --query http://localhost/eye/reasoning/socrates/socrates-query.n3

GET http://localhost/eye/reasoning/socrates/socrates.n3 SC=3
GET http://localhost/eye/reasoning/socrates/socrates-query.n3 SC=1
networking 5 [msec cputime] 24 [msec walltime]
@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#>.
@prefix : <http://example.org/socrates#>.

:Socrates a :Human.
:Socrates a :Mortal.

reasoning 0 [msec cputime] 0 [msec walltime]
#2022-04-14T16:53:38.202Z in=4 out=2 ent=3 step=6 brake=2 inf=23148 sec=0.037 inf/sec=625622
#ENDS

2022-04-14T16:53:38.202Z in=4 out=2 ent=3 step=6 brake=2 inf=23148 sec=0.037 inf/sec=625622
```
but via eyeserver nothing comes back
```
$ curl "http://localhost:8000/?data=http://localhost/eye/reasoning/socrates/socrates.n3&query=http://localhost/eye/reasoning/socrates/socrates-query.n3"

```
@RubenVerborgh any idea?

**Comment by RubenVerborgh** [comment]
Oh yes, we are very likely deliberately blocking `localhost` access for security reasons 🙂

**Comment by RubenVerborgh** [comment]
Yeah we are: https://github.com/RubenVerborgh/EyeServer/blob/36b3e87fd263af122c8af6ab54ea681eeaab6734/lib/eye.js#L93

Can make this into a flag.

**Comment by josd** [comment]
That would be a nice option indeed!

**Comment by MartyStache** [comment]
awesome! thanks!
